packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedMessageEmbedFields.ts

```diff
@@ -114,17 +114,17 @@ export class PaginatedMessageEmbedFields extends PaginatedMessage {
 	}
 
 	private generatePages(): void {
-		const template = this.embedTemplate instanceof EmbedBuilder ? this.embedTemplate.toJSON() : this.embedTemplate;
+		const template = this.embedTemplate;
 		for (let i = 0; i < this.totalPages; i++) {
-			const clonedTemplate = new EmbedBuilder(template);
-			const fieldsClone = this.embedTemplate.data.fields ?? [];
-			clonedTemplate.data.fields = [];
+			const fields = isNullishOrEmpty(template.fields) ? [] : [...template.fields];
+			const embed = new EmbedBuilder(template);
```

Comment:
Why are these flipped, unflip them. Also revert the names to `clonedTemplate` and `fieldsClone`. There was no reason to rename them.